.github/workflows/golangci-lint.yml

```diff
@@ -1,8 +1,7 @@
 name: golangci-lint
 on:
   push:
-    branches:
-      - main
+    branches: [ main, releases-1.0.x ]
   pull_request:
```

Comment:
So I'm actually curious about why the lint action fires on push and not when a PR is created? Should we move this to when a pull request is created to reduce the # of lint jobs fired?

Reply:
It probably needs to run on each commit?

Reply:
That's right. Push event is every commit that gets pushed. Pull request event is when you open a PR, when you synchronize a PR, when you close a PR, and a bunch of other activities that can happen for a PR. Though only some of those activities are enabled by default.